pkg/morph/subscriber/subscriber.go

```diff
@@ -188,3 +199,29 @@ func New(ctx context.Context, p *Params) (Subscriber, error) {
 
 	return sub, nil
 }
+
+// awaitHeight checks if remote client has least expected block height and
+// returns error if it is not reached that height after timeout duration.
+// This function is required to avoid connections to unsynced RPC nodes, because
+// they can produce events from the past that should not be processed by
+// NeoFS nodes.
+func awaitHeight(wsClient *client.WSClient, startFrom uint32, timeout time.Duration) error {
+	if startFrom == 0 {
+		return nil
+	}
+
+	for ch := time.After(timeout); ; {
+		select {
+		case <-ch:
+			return fmt.Errorf("could not init ws client: didn't reach expected height %d", startFrom)
+		default:
+		}
+		height, err := wsClient.GetBlockCount()
+		if err != nil {
+			return fmt.Errorf("could not get block height: %w", err)
+		} else if height >= startFrom {
+			return nil
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
isn't it too frequent?

Reply:
Removed in https://github.com/nspcc-dev/neofs-node/pull/798/commits/b52d3fcbf6d8800b0952b784a1abba50134ac2a7